### PR TITLE
8253751: Dependencies of automatic modules are not propagated through module layers

### PR DESCRIPTION
**Pull request by AlanBateman**
This is a corner case that arises when creating a Configuration for a child module layer. If an explicit module in the child configuration reads an automatic module in a parent configuration then it should read all automatic modules in the parent configurations. Unfortunately, this read edge wasn't created for the case that the child layer does not contain any automatic modules.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8253751](https://bugs.openjdk.java.net/browse/JDK-8253751): Dependencies of automatic modules are not propagated through module layers

### Reviewers
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1391/head:pull/1391`
`$ git checkout pull/1391`

### CODE REVIEW

File: src/java.base/share/classes/java/lang/module/Resolver.java [hunk 1]
```diff
@@ -572,6 +572,15 @@ private static String toHexString(byte[] ba) {
                         assert requires.modifiers().contains(Modifier.STATIC);
                         continue;
                     }
+
+                    // m2 is automatic module in parent configuration => m1 reads
+                    // all automatic modules that m2 reads.
+                    if (m2.descriptor().isAutomatic()) {
+                        m2.reads()
+                                .stream()
```
Comment (mlchung) [inline src/java.base/share/classes/java/lang/module/Resolver.java:580]: Nit: not sure if this formatting is intentionally or you meant `m2.reads().stream()` in one line (which I like better).

**Review by mlchung — APPROVED** [review]
Looks good.

File: src/java.base/share/classes/java/lang/module/Resolver.java (hunk 1 above)
Comment (AlanBateman) [reply src/java.base/share/classes/java/lang/module/Resolver.java]: Thanks, I suspect I just hit Return and the IDE formatted it. Fixed now.

**Review by mlchung — APPROVED** [review]

**Pull request closed without merging**